frontend/model/notifications/vuexModule.js

```diff
@@ -0,0 +1,104 @@
+import type {
+  Notification,
+  VuexModuleContext
+} from './types.flow.js'
+
+import sbp from '~/shared/sbp.js'
+import './selectors.js'
+import { age, isNew, isOlder } from './utils.js'
+import * as keys from './mutationKeys.js'
+
+const actions = {
+  async login (context: VuexModuleContext, user: Object) {
+    const notifications = await sbp('gi.db/notifications/load', user.username)
+
+    if (notifications && notifications.length) {
+      context.commit(keys.SET_NOTIFICATIONS, notifications)
+    }
+  },
+  async logout (context: VuexModuleContext) {
+    if (context.rootState.loggedIn) {
+      // Make sure to save the notifications *before* clearing them.
+      await sbp('gi.notifications/save')
```

Comment:
Isn't this missing the username?

Reply:
Since there is always a single logged-in username per session, I've made `sbp('gi.notifications/save')` just look it up by itself, so that there should be no need to specify it.